builder/optimizations.cc

```diff
@@ -24,25 +25,8 @@ namespace slinky {
 namespace {
 
 bool is_copy(expr in, var out, expr& offset) {
-  static var x(0), dx(1), negative_dx(2);
-  static expr patterns[] = {
-      x, x + dx, x - negative_dx,
-      // TODO: we could also handle scaling of x by multiplying the stride.
-  };
-
-  symbol_map<expr> matches;
-  for (const expr& p : patterns) {
-    matches.clear();
-    if (match(p, in, matches) && match(*matches[x], out)) {
-      offset = 0;
-      // We found a pattern that is a copy. We don't care which one, we just need to look at the matches we have.
-      if (matches[dx]) offset = *matches[dx];
-      if (matches[negative_dx]) offset = -*matches[negative_dx];
-      return true;
-    }
-  }
-
-  return false;
+  offset = simplify(in - out);
```

Comment:
It's unclear to me why this function is called ```is_copy```? Could you please add some comments?